ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -0,0 +1,209 @@
+
+Web apps can request for the clipboard-read permissions by performing a read operation using one of [read](https://w3c.github.io/clipboard-apis/#dom-clipboard-read) or [readText](https://w3c.github.io/clipboard-apis/#dom-clipboard-readtext) methods of the [Async clipboard API](https://w3c.github.io/clipboard-apis/#async-clipboard-api). Reading the clipboard before starting to listen to clipboard changes (to build the intial state) is a typical sequence of actions in web apps that would use the "clipboardchange" event.
+
+In future, we may have additional ways of prompting the user for permissions - 1) By explicitly requesting for "clipboard-read" permission, the API for this is still under discussion (https://github.com/w3c/permissions/issues/158). 2) The user can be prompted for permissions as soon as the "addEventListener" method is called with "clipboardchange" in case the permissions are not already granted. This is still open for discussion as it is not a common pattern to prompt user for permissions when attaching event listeners. 3) The user can be prompted for permissions just before the browser dispatches the "clipboardchange" event. This way, the permissions prompt would appear only when required by the browser however web authors won't have control over when the prompt would be triggered which might not be desirable.  
+
+
+##### Pros
+1. Since clipboard can be changed by a user action, the information regarding when the clipboard has changed can be considered a user private data. Having the permissions check ensures that this privacy sensitive information is protected by user consent.
+2. This approach aligns with the security model for other clipboard APIs.
+
+##### Cons
+1. This approach imposes a small restriction on web authors as they have to call a clipboard read API method before starting to listen for the clipboardchange event.
+
+##### 5.1.1.1 Permissions policy integration in iframe and fencedframe
+
+Listening to "clipboardchange" event within a cross-origin iframe can be enabled with [Permissions Policy](https://www.w3.org/TR/permissions-policy) which allows for selectively enabling and disabling specific browser features and APIs. Specifically, "clipboard-read" permission needs to be passed to the "allow" attribute of the iframe, similar to how this is required for accessing async clipboard read APIs within a cross-origin iframe.
+
+```html
+<iframe
+    src="https://other-domain.com/will-monitor-clipboard.html"
+    allow="clipboard-read;"
+>
+</iframe>
+```
+
+This event is not accessible within a "fencedframe" since the clipboard and the clipboardchange event could be used as a communication channel between the host and the fencedframe, so constitute a privacy threat.
```

Comment:
Fenced frames are a Chromium-only concept and other browsers may not work this way. Do you just mean to say "cross-origin iframe" here? If so, please update the term and how to handle cross-origin iframes can be tracked as an open issue.

Reply:
Moved to open issues